src/main/java/hcfsfuse/fuse/HCFSFuseFileSystem.java

```diff
@@ -97,6 +104,52 @@ public HCFSFuseFileSystem(FileSystem fs, FuseOptions fuseOptions,
     mOpenFiles = new IndexedSet<>(ID_INDEX, PATH_INDEX);
   }
 
+  @Override
+  public int chown(String path, @uid_t long uid, @gid_t long gid) {
+
+    try {
+      final Path turi = mPathResolverCache.getUnchecked(path);
+
+      String userName = "";
+      if (uid != ID_NOT_SET_VALUE && uid != ID_NOT_SET_VALUE_UNSIGNED) {
+        userName = AlluxioFuseUtils.getUserName(uid);
+        if (userName.isEmpty()) {
+          // This should never be reached
+          LOG.error("Failed to get user name from uid {}", uid);
+          return -ErrorCodes.EINVAL();
+        }
+      }
+
+      String groupName = "";
+      if (gid != ID_NOT_SET_VALUE && gid != ID_NOT_SET_VALUE_UNSIGNED) {
+        groupName = AlluxioFuseUtils.getGroupName(gid);
+        if (groupName.isEmpty()) {
+          // This should never be reached
+          LOG.error("Failed to get group name from gid {}", gid);
+          return -ErrorCodes.EINVAL();
+        }
+      } else if (!userName.isEmpty()) {
+        groupName = AlluxioFuseUtils.getGroupName(userName);
+      }
+
+      if (userName.isEmpty() && groupName.isEmpty()) {
+        // This should never be reached
+        LOG.info("Unable to change owner and group of file {} when uid is {} and gid is {}", path,
+                userName, groupName);
+      } else if (userName.isEmpty()) {
+        LOG.info("Change group of file {} to {}", path, groupName);
+        mFileSystem.setOwner(turi, userName, groupName);
```

Comment:
if `username` is empty , what will happen?

Reply:
This is normal. One of the users or groups is allowed to be empty in chown

Reply:
I mean that if users or groups is `null`, the original username or group name remains unchanged. But `""` is empty, so it mean change username or groupname to `""` empty, any thoughts?